spring-app/src/main/kotlin/pl/starchasers/up/security/JwtTokenFilter.kt

```diff
@@ -1,37 +1,104 @@
 package pl.starchasers.up.security
 
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
-import org.springframework.security.core.GrantedAuthority
-import org.springframework.security.core.authority.SimpleGrantedAuthority
 import org.springframework.security.core.context.SecurityContextHolder
 import org.springframework.web.filter.GenericFilterBean
+import org.springframework.web.util.WebUtils
 import pl.starchasers.up.service.JwtTokenService
-import pl.starchasers.up.service.ROLE_KEY
+import pl.starchasers.up.util.SetCookieHeaderValueBuilder
+import java.time.Duration
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.time.temporal.TemporalUnit
 import javax.servlet.FilterChain
 import javax.servlet.ServletRequest
 import javax.servlet.ServletResponse
 import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
 
 class JwtTokenFilter(
     private val tokenService: JwtTokenService
 ) : GenericFilterBean() {
 
     override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain) {
-        val token = (request as HttpServletRequest).getHeader("Authorization")
+        val accessTokenString = WebUtils.getCookie(
+            (request as HttpServletRequest), JwtTokenService.ACCESS_TOKEN_COOKIE_NAME
+        )?.value
+        val refreshTokenString = WebUtils.getCookie(
+            request, JwtTokenService.REFRESH_TOKEN_COOKIE_NAME
+        )?.value
 
-        try {
-            val claims = tokenService.parseToken(token.removePrefix("Bearer "))
+        var newAccessTokenString: String?
+        val newRefreshTokenString: String?
 
-            val authorities = mutableListOf<GrantedAuthority>()
-            authorities.add(SimpleGrantedAuthority(Role.USER.roleString()))
-            if (Role.valueOf(claims[ROLE_KEY] as String) == Role.ADMIN) authorities.add(SimpleGrantedAuthority(Role.ADMIN.roleString()))
+        try {
+            newRefreshTokenString = refreshRefreshToken(refreshTokenString)
+            newAccessTokenString = refreshAccessToken(accessTokenString, newRefreshTokenString)
+            newAccessTokenString = issueAccessToken(newAccessTokenString, newRefreshTokenString)
```

Comment:
I'm nitpicking here, but we could do without that double assignment. Both conditions can be merged into a single method `accessToken == null || isCloseTo(...)`

Reply:
This can be addressed in the future